lenskit-eval/src/main/java/org/lenskit/eval/traintest/TestUser.java

```diff
@@ -143,7 +155,10 @@ public LongSet getSeenItems() {
      */
     public Long2DoubleMap getTestRatings() {
         if (testRatings == null) {
-            testRatings = Ratings.userRatingVector(testHistory);
+            Predicate<Entity> predicate = Entities.typePredicate(CommonTypes.RATING);
+            Function<Entity, Rating> targetViewClass = Entities.projection(Rating.class);
+            ImmutableList<Rating> list = FluentIterable.from(trainHistory).filter(predicate).transform(targetViewClass).toList();
```

Comment:
Here's the problem that's causing test failure. This should grab the `testHistory`, not the `trainHistory`.